### DISCUSSION TITLE
Merge pull request loop across forked repos

### DISCUSSION THREAD

**Discussion by crizmo** [community/community#62046] [Pull Requests]
### Select Topic Area

Question

### Body

Lets say i have a organization [ I'm the owner of it ] 
now i have a repository inside that organization which i forked on my personal account 

so now when i commit something to the forked repository
i can make a pull request across forks on the organization repository

so it shows that my forked repository is 1 commit ahead of the organization repository
when i merge the pull request on the organization repository it shows that my personal repository is 1 commit behind the organization repository which is basically the merged commit

So now i can again make a pull request on my personal forked repository which then becomes 1 commit ahead of the organization repository and it becomes a loop

Is this supposed to happen ?

**Comment by nirav-gajera** [comment]
Yes, this is expected behavior. When you fork a repository, you create a copy of the repository that is independent from the original repository. This means that you can make changes to your fork without affecting the original repository. You can then submit a pull request to the original repository to merge your changes.

When you merge a pull request, the original repository is updated to include your changes. This means that your fork will now be one commit behind the original repository. You can then continue to make changes to your fork and submit new pull requests.

This can create a loop, where you are constantly making changes to your fork and merging those changes into the original repository. This is perfectly fine, and it is a common way to contribute to open source projects.

**Comment by crizmo** [comment]
Alrighty , got more info on this 
and adding branch rule [ Require linear history ]  on the main repo prevents this 
basically what i wanted 
Ty
